addons/project/models/project.py

```diff
@@ -1800,6 +1802,10 @@ def create(self, vals_list):
             # user_ids change: update date_assign
             if vals.get('user_ids'):
                 vals['date_assign'] = fields.Datetime.now()
+                if not vals.get('project_id'):
+                    user_ids = self._fields['user_ids'].convert_to_cache(vals.get('user_ids', []), self)
+                    if self.env.user.id not in user_ids:
+                        vals['user_ids'] = [Command.set(list(user_ids) + [self.env.user.id])]
```

Comment:
We should take `default_project_id ` into consideration too. As I tested in the quick create in a project, and I was always added in the assignees which is not the desired behavior.
```suggestion
                if not project_id:
                    user_ids = self._fields['user_ids'].convert_to_cache(vals.get('user_ids', []), self)
                    if self.env.user.id not in user_ids:
                        vals['user_ids'] = [Command.set(list(user_ids) + [self.env.user.id])]
```